### ISSUE TITLE
FR: Bundle-size checks for ESM code

### ISSUE THREAD

**Issue by rsimha** [ampproject/amphtml#27703]
With the addition of ESM builds, we'd like the ability to run bundle-size checks for two flavors of minified code that are compiled in parallel on two separate Travis Jobs.

This FR is a tracking issue for the work that needs to be done towards this on the `amphtml` repo. Feel free to split off a separate issue for the `amp-github-apps` stuff.

**Comment by rsimha** [comment]
Co-assigning to Erwin, in case this needs accompanying changes to the ESM code (e.g. the proposed `--module` and `--nomodule` flags for `gulp dist`).

**Comment by danielrozenberg** [comment]
This is indeed blocked on `--module`/`--nomodule` flags

**Comment by erwinmombay** [comment]
based on our discussions its isn't blocked anymore when we start using the new `gulp release` task?

**Comment by danielrozenberg** [comment]
No, the we won't be using `gulp release` in our Travis builds, and bundle-size currently cannot be split between two different jobs - so we need to be able to compile both ESM and non-ESM in the same Travis shard

**Comment by rsimha** [comment]
Fixed by #31025
